add-on/src/lib/ipfs-request.js

```diff
@@ -43,7 +49,7 @@ function createRequestModifier (getState, dnsLink, ipfsPathValidator) {
       }
       // Detect valid /ipfs/ and /ipns/ on any site
       if (ipfsPathValidator.publicIpfsOrIpnsResource(request.url)) {
-        return redirectToCustomGateway(request.url, state.gwURL)
+        return redirectToIpfsTransport(request.url, state, featureDetector)
```

Comment:
I don't know what a `IpfsTransport` is, how about just `redirectToGateway` (could be custom or public)